### PR TITLE
first implementation

### PR DESCRIPTION
**Pull request by itajaja**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,22 @@
+The MIT License (MIT)
+
+Copyright (c) 2015 4Catalyzer
```
Comment (yodaiken) [inline LICENSE]: !

Comment (itajaja) [reply LICENSE]: oh gosh time flies.

File: tqp/.threading_utils.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    def run(self):
+        while not self.finished.wait(self.interval):
+            self.function(*self.args, **self.kwargs)
+
+    def __enter__(self):
+        self.start()
+        return self
+
+    def __exit__(self, *args):
+        self.cancel()
```
Comment (taion) [inline tqp/.threading_utils.py]: do we need to handle error cases here?

File: tqp/topic_queue_poller.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+import logging
+
+import boto3
+
+from .threading_utils import Interval
+
+# -----------------------------------------------------------------------------
+
+
+def _jsonize_dictionary(dictionary):
```
Comment (taion) [inline tqp/topic_queue_poller.py]: `jsonify_dictionary`?

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    url='https://github.com/4Catalyzer/tqp',
+    author="Giacomo Tagliabue",
+    author_email='giacomo@gmail.com',
+    license='MIT',
+    classifiers=(
+        'Development Status :: 3 - Alpha',
+        'License :: OSI Approved :: MIT License',
+        'Operating System :: OS Independent',
+        'Programming Language :: Python',
+        'Programming Language :: Python :: 2',
```
Comment (taion) [inline setup.py]: i don't think this is python 2 compatible

Comment (itajaja) [reply setup.py]: yeah right. Are we fine dropping py2 compat?

File: tqp/.threading_utils.py (hunk 2 above)
Comment (itajaja) [reply tqp/.threading_utils.py]: if `self.cancel()` throws? not sure what to do beside ignoring, I'd rather let it fail

Comment (taion) [reply tqp/.threading_utils.py]: no i mean if the thing inside the context manager throws

File: setup.py (hunk 4 above)
Comment (taion) [reply setup.py]: fine by me

File: tqp/.threading_utils.py (hunk 2 above)
Comment (itajaja) [reply tqp/.threading_utils.py]: uh? isn't that in the main function flow?

**Pull request merged**